dev-tools/mage/crossbuild.go

```diff
@@ -179,8 +194,13 @@ func (b GolangCrossBuilder) Build() error {
 
 	mountPoint := filepath.ToSlash(filepath.Join("/go", "src", repoInfo.RootImportPath))
 	workDir := mountPoint
+	buildCmd := "build/mage-linux-amd64"
 	if repoInfo.SubDir != "" {
 		workDir = filepath.ToSlash(filepath.Join(workDir, repoInfo.SubDir))
+		if b.XPack {
+			workDir = filepath.ToSlash(filepath.Join(workDir, "x-pack", repoInfo.SubDir))
+			buildCmd = filepath.Join("..", "..", repoInfo.SubDir, buildCmd)
```

Comment:
If you do add some kind of target/work dir parameter to GolangCrossBuilder then I think you can use `filepath.Rel` here to get a relative path from the current working directory to the target dir.